ddtrace/contrib/trace_utils.py

```diff
@@ -276,6 +305,40 @@ def set_http_meta(
     if retries_remain is not None:
         span._set_str_tag(http.RETRIES_REMAIN, str(retries_remain))
 
+    if tracer is None or tracer._gateway is None:  # tracer has a gateway only when AppSec is enabled
+        return
+    gateway = tracer._gateway
+    if gateway.needed_address_count == 0:
+        return
+
+    data = {}
+    _add_if_needed(gateway, data, raw_uri, _Addresses.SERVER_REQUEST_URI_RAW)
+    _add_if_needed(gateway, data, status_code, _Addresses.SERVER_RESPONSE_STATUS, str)  # make sure it's a string
+    _add_if_needed(gateway, data, query_object, _Addresses.SERVER_REQUEST_QUERY, format_query_object)
+    _add_if_needed(gateway, data, request_cookies, _Addresses.SERVER_REQUEST_COOKIES)
+    _add_if_needed(gateway, data, request_headers, _Addresses.SERVER_REQUEST_HEADERS_NO_COOKIES, format_request_headers)
+    req_cookies_key = _Addresses.SERVER_REQUEST_HEADERS_NO_COOKIES.value
+    if req_cookies_key in data:
+        data[req_cookies_key] = _no_cookies(data[req_cookies_key])
+
+    _add_if_needed(
+        gateway, data, response_headers, _Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES, format_response_headers
+    )
+    res_cookies_key = _Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES.value
+    if res_cookies_key in data:
+        data[res_cookies_key] = _no_cookies(data[res_cookies_key])
+
+    _add_if_needed(gateway, data, request_body, _Addresses.SERVER_REQUEST_BODY, format_request_body)
+    _add_if_needed(
+        gateway, data, request_path_params, _Addresses.SERVER_REQUEST_PATH_PARAMS, format_request_path_params
+    )
+
+    if len(data.keys()) == 0:
+        return
+
+    store = tracer._current_context_store()
```

Comment:
Again, tracer/span are being overridden to be used as a generalized context API. We've agreed that it's ok for now but we'll want to rework this in the future.

We should update the PR description to indicate that there are future changes required

Reply:
I updated the description, let me know what you think :)